strawberryfields/decompositions.py

```diff
@@ -47,6 +47,25 @@ def takagi(N, tol=1e-13, rounding=13):
     if np.linalg.norm(N - np.transpose(N)) >= tol:
         raise ValueError("The input matrix is not symmetric")
 
+    N = np.real_if_close(N)
+
+    if np.allclose(N, np.zeros(N.shape)):
```

Comment:
(Let me know if I'm missing something here!)
```suggestion
    if np.allclose(N, 0):
```

Reply:
Here `N` is a matrix, you want to compare it with the zero matrix of the same dimension.

Reply:
I think this is correct as is.

Reply:
Right! The reason was, that with the suggestion, numpy's broadcasting will happen anyways, essentially comparing a zero matrix even if writing `0`. My initial thought was that this would perform significantly better.

```python
import numpy as np

shape1 = 1000

a = np.zeros((shape1, shape1))
b = np.zeros((shape1, shape1))
%timeit -n 500 np.allclose(a, b)
6.8 ms ± 400 µs per loop (mean ± std. dev. of 7 runs, 500 loops each)
```
```python
%timeit -n 500 np.allclose(a, 0)
1.88 ms ± 125 µs per loop (mean ± std. dev. of 7 runs, 500 loops each)
```
I had to, however, increasing the size of `shape1` to get a real difference.

Reply:
So overall, I'm fine with whichever way! Having `0` reads slightly better and seems to perform better for large matrices, but I have no strong preference.